app/components/post_list/post_list.test.tsx

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {act, fireEvent} from '@testing-library/react-native';
+import React from 'react';
+import {DeviceEventEmitter, Platform} from 'react-native';
+
+import * as localPostFunctions from '@actions/local/post';
+import * as postFunctions from '@actions/remote/post';
+import {Events, Screens} from '@constants';
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import PostList, {exportedForTesting} from './post_list';
+
+jest.mock('@components/post_list/post', () => 'Post');
+jest.mock('@components/post_list/thread_overview', () => 'ThreadOverview');
+jest.mock('@components/post_list/more_messages', () => 'MoreMessages');
+jest.mock('@components/post_list/combined_user_activity', () => 'CombinedUserActivity');
+jest.mock('@components/post_list/scroll_to_end_view', () => 'ScrollToEndView');
+jest.mock('@actions/remote/post', () => {
+    return {
+        fetchPosts: jest.fn(),
+        fetchPostThread: jest.fn(),
+    };
+});
+jest.mock('@actions/local/post', () => ({
+    removePost: jest.fn(),
+}));
```

Comment:
Mocking these helps with fetchPosts, and the other 2 functions from actually being called since those functions are in useCallback.  Doing jest.spyOn is not enough to mock those functions inside useCallback (because they have already been defined?).